isort/sorting.py

```diff
@@ -54,6 +54,7 @@ def module_key(
 def section_key(
     line: str,
     case_sensitive: bool,
+    honor_case_in_force_sorted_sections: bool,
     order_by_type: bool,
     force_to_top: List[str],
     lexicographical: bool = False,
```

Comment:
This function is getting rather a lot of arguments now. This is only called in one place, and all arguments are part of the `config`. Should this function be changed to just take a `config` argument, like `module_key` above also does?

`module_key` also needs to be able to override the config since it's called in some different contexts. That's not the case for this function. Is there still a good reason to keep any arguments separate?

And if you think it's a good idea to fix this, should I do it in this pull request or a separate one? (I think probably a separate one.)

Reply:
I think your instinct is correct that it does make sense to refactor it to use the immutable config directly. I have a preference for more smaller PRs as they are easier to review, but the project has no official stance and I will merge either as long as the tests pass and the changeset looks good. Honestly, just thrilled to have the contributions, thank you!